description: 'USB: core: fix out-of-bounds access bug in usb_get_bos_descriptor()'
references:
- https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2017-16535
- https://github.com/torvalds/linux/commit/1c0edc3633b56000e18d82fc241e3995ca18a69e
- https://groups.google.com/d/msg/syzkaller/tzdz2fTB1K0/OvjIgLSTAgAJ
- https://usn.ubuntu.com/usn/usn-3485-1
- https://usn.ubuntu.com/usn/usn-3485-2
- https://usn.ubuntu.com/usn/usn-3485-3
- https://usn.ubuntu.com/usn/usn-3507-1
- https://usn.ubuntu.com/usn/usn-3754-1
- https://ubuntu.com/security/notices/USN-3485-1
- https://ubuntu.com/security/notices/USN-3485-2
- https://ubuntu.com/security/notices/USN-3485-3
- https://ubuntu.com/security/notices/USN-3507-1
- https://ubuntu.com/security/notices/USN-3754-1
- https://www.cve.org/CVERecord?id=CVE-2017-16535
comments:
  debian/bwh: 'Introduced in 3.2 by commit 3148bf041d16 "usbcore: get BOS descriptor
    set"'
reporters:
- Andrey Konovalov
introduced-by:
  mainline: [3148bf041d169a083aa31bd69bedd5bfb7ffe215]
fixed-by:
  aosp/3.10: [ea90468b7c4d33a4bd63adafd114dd04e1f7f278]
  aosp/3.18: [7c1c88e160ed14dfb02cb35369e27abe01eb2ca5]
  cip/4.4: [9dff499d822660c6dbb2a407a7d85be26f87da07]
  cip/4.4-rt: [9dff499d822660c6dbb2a407a7d85be26f87da07]
  cip/4.4-st: [9dff499d822660c6dbb2a407a7d85be26f87da07]
  debian/buster-security: ['version:debian/4.13.10-1']
  debian/jessie-security: ['version:debian/3.16.51-1']
  debian/stretch-security: ['version:debian/4.9.65-1']
  mainline: [1c0edc3633b56000e18d82fc241e3995ca18a69e]
  stable/3.16: [6514189e83d470af2f35735038c1b096410ab98d]
  stable/3.18: [7c1c88e160ed14dfb02cb35369e27abe01eb2ca5]
  stable/3.2: [7c27b82fad16d2804c7c8405316a636f57edeabd]
  stable/4.1: [e4e541386c2e0221a2325fc83b18ddf683db1f29]
  stable/4.13: [775462fd5c5378957e5165c898e2fa532ebf4234]
  stable/4.4: [9dff499d822660c6dbb2a407a7d85be26f87da07]
  stable/4.9: [9d13d3e05be29056eeab610d9ad26b04c9231a04]
  ubuntu/focal: ['version:5.4.0-9.12']
  ubuntu/xenial: [a0d346c82ade624433c76d270aa33936f543c5cf]
